//
// PROJECT:         Aspia
// FILE:            category/category_connection.proto
// LICENSE:         GNU Lesser General Public License 2.1
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message Connection
{
    enum Protocol
    {
        PROTOCOL_UNKNOWN = 0;
        PROTOCOL_TCP     = 1;
        PROTOCOL_UDP     = 2;
    }

    enum State
    {
        STATE_UNKNOWN     = 0;
        STATE_CLOSED      = 1;
        STATE_LISTENING   = 2;
        STATE_SYN_SENT    = 3;
        STATE_SYN_RCVD    = 4;
        STATE_ESTABLISHED = 5;
        STATE_FIN_WAIT1   = 6;
        STATE_FIN_WAIT2   = 7;
        STATE_CLOSE_WAIT  = 8;
        STATE_CLOSING     = 9;
        STATE_LAST_ACK    = 10;
        STATE_TIME_WAIT   = 11;
        STATE_DELETE_TCB  = 12;
    }

    message Item
    {
        string process_name   = 1;
        Protocol protocol     = 2;
        string local_address  = 3;
        string remote_address = 4;
        uint32 local_port     = 5;
        uint32 remote_port    = 6;
        State state           = 7;
    }

    repeated Item item = 1;
}
